clang/test/CIR/CodeGen/call-side-effect.cpp

```diff
@@ -0,0 +1,25 @@
+// RUN: %clang_cc1 -std=c++20 -triple x86_64-unknown-linux-gnu -fclangir -emit-cir %s -o %t.cir
+// RUN: FileCheck --input-file=%t.cir --check-prefix=CIR %s
+// RUN: %clang_cc1 -std=c++20 -triple x86_64-unknown-linux-gnu -fclangir -emit-llvm %s -o %t.ll
+// RUN: FileCheck --input-file=%t.ll --check-prefix=LLVM %s
+
+[[gnu::pure]] int pure_func(int x);
+[[gnu::const]] int const_func(int x);
+
+int test(int x) {
+  int y1 = pure_func(x);
+  int y2 = const_func(x);
+  return y1 + y2;
+}
+
+// CIR-LABEL: @_Z4testi
+// CIR:   %{{.+}} = cir.call @_Z9pure_funci(%{{.+}}) : (!s32i) -> !s32i sideeffect(pure)
```

Comment:
We need a test that leads to `sideeffect(all)`

Reply:
CIRGen does not emit `side_effect(all)`. Added a test from CIR `side_effect(all)` to LLVMIR attributes.

Reply:
If CIRGen doesn't emit it, do we need it at all?

Reply:
Well I attach the attribute to call operations as a default-valued attribute, with `all` as the default value.

Reply:
Gotcha!